packages/cardano-services/src/Handle/TypeOrmHandleProvider.ts

```diff
@@ -0,0 +1,58 @@
+import { HandleEntity } from '@cardano-sdk/projection-typeorm';
+import {
+  HandleProvider,
+  HandleResolution,
+  Point,
+  ProviderError,
+  ProviderFailure,
+  ResolveHandlesArgs
+} from '@cardano-sdk/core';
+import { In } from 'typeorm';
+import { TypeormProvider, TypeormProviderDependencies } from '../util/TypeormProvider';
+
+export type TypeOrmHandleProviderDependencies = TypeormProviderDependencies;
+
+const handleFields = ['cardanoAddress', 'handle', 'hasDatum', 'policyId'] as const;
+
+type HandleFields = typeof handleFields[number];
+type PartialHandleEntity = {
+  [k in HandleFields]: Required<HandleEntity>[k];
+};
+
+const handleSelect = <{ [k in HandleFields]: true }>Object.fromEntries(handleFields.map((_) => [_, true]));
+
+export const emptyStringHandleResolutionRequestError = () =>
+  new ProviderError(ProviderFailure.BadRequest, undefined, "Empty string handle can't be resolved");
+
+export class TypeOrmHandleProvider extends TypeormProvider implements HandleProvider {
+  constructor(deps: TypeOrmHandleProviderDependencies) {
+    super('TypeOrmHandleProvider', deps);
+  }
+
+  async resolveHandles(args: ResolveHandlesArgs) {
+    const { handles } = args;
+
+    for (const handle of handles) if (!handle) throw emptyStringHandleResolutionRequestError();
+
+    return this.withDataSource(async (dataSource) => {
+      const queryRunner = dataSource.createQueryRunner();
+      const query = 'SELECT hash, slot FROM block WHERE slot = (SELECT MAX(slot) FROM block)';
```

Comment:
Should this query be saved to a queries file within this folder or another appropriate one?

Reply:
This is our existing standard, so it's best to not  deviate here, but rather in a separate complete refactor commit. Having said that, small queries like this work well inline as we have the context in the same file, plus there's not a demand for this query to be reused.

Reply:
From @rhyslbw 's comment I would say this doesn't need to be changed

Reply:
Yea, I agree.